src/Bridge/Elasticsearch/DataProvider/CollectionDataProvider.php

```diff
@@ -0,0 +1,121 @@
+use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\Pagination;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use Elasticsearch\Client;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+/**
+ * Collection data provider for Elasticsearch.
+ *
+ * @experimental
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+final class CollectionDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    private $client;
+    private $documentMetadataFactory;
+    private $identifierExtractor;
+    private $denormalizer;
+    private $pagination;
+    private $collectionExtensions;
+
+    /**
+     * @param FullBodySearchCollectionExtensionInterface[] $collectionExtensions
+     */
+    public function __construct(Client $client, DocumentMetadataFactoryInterface $documentMetadataFactory, IdentifierExtractorInterface $identifierExtractor, DenormalizerInterface $denormalizer, Pagination $pagination, iterable $collectionExtensions = [])
+    {
+        $this->client = $client;
+        $this->documentMetadataFactory = $documentMetadataFactory;
+        $this->identifierExtractor = $identifierExtractor;
+        $this->denormalizer = $denormalizer;
+        $this->pagination = $pagination;
+        $this->collectionExtensions = $collectionExtensions;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(string $resourceClass, ?string $operationName = null, array $context = []): bool
+    {
+        try {
+            $this->documentMetadataFactory->create($resourceClass);
+        } catch (IndexNotFoundException $e) {
+            return false;
+        }
+
+        try {
+            $this->identifierExtractor->getIdentifierFromResourceClass($resourceClass);
+        } catch (NonUniqueIdentifierException $e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCollection(string $resourceClass, ?string $operationName = null, array $context = [])
+    {
+        $documentMetadata = $this->documentMetadataFactory->create($resourceClass);
+        $body = [];
+
+        foreach ($this->collectionExtensions as $collectionExtension) {
+            $body = $collectionExtension->applyToCollection($body, $resourceClass, $operationName, $context);
+        }
+
+        if (!isset($body['query']) && !isset($body['aggs'])) {
+            $body = array_merge($body, [
+                'query' => [
+                    'match_all' => new \stdClass(),
+                ],
+            ]);
+        }
+
+        $limit = $body['size'] ?? $this->pagination->getLimit($resourceClass, $operationName);
+        $offset = $body['offset'] ?? $this->pagination->getOffset($resourceClass, $operationName);
+
+        if (!isset($body['size'])) {
+            $body = array_merge($body, ['size' => $limit]);
```

Comment:
merges can be avoided like `$limit = $body['size'] = $body['size'] ?? 'val'`, or just `$body['size'] = $limit;` here :thinking:

Reply:
Wow! Where was my mind? All merges can be avoided in this method... Thx.